docs/designs/security.md

```diff
@@ -0,0 +1,19 @@
+# Security Controller
+
+The security controller deals with everything related to the opensearch-security plugin. This entails two areas:
+
+* Configuring TLS for nodes
+* Manging the securityconfig
+
+The controller is configured via the `security` object in the cluster spec. If no config is provided the controller will fall back to the demo certificates and configuration that is included with the opensearch docker image.
+
```

Comment:
I would add that: 
1. the state (self signed certificates, root CA, etc) is stored in K8s using K8s Secrets API
2. If you want to read more about Generating Certs for OpenSearch, read here: https://opensearch.org/docs/latest/security-plugin/configuration/generate-certificates/
3. I would list what you keys you configure for OpenSearch Security plugin in this README to save people from reading the code to figure that one out
4. The certificates have two options: Bring your own; Operator generates it for you. Also would specify how to select between the modes in the CRD (which section to change)

Reply:
Added more details to the design